### PR TITLE
Re-writing Grip Build to be Strongly Typed

### PR DESCRIPTION
**Pull request by JLLeitschuh**
[//]: # (Please ensure that the "Allow edits from maintainers" checkbox is checked. Thanks!)

Work in progress

### CODE REVIEW

File: ui/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+                    "-Dprism.order=sw",
+                    "-Dprism.text=t2k"
+            )
+        }
+    }
+}
+
+idea {
+    module {
+        scopes["PROVIDED"]?.get("plus")?.add(ideProviderConfiguration)
```
Comment (JLLeitschuh) [inline ui/build.gradle.kts]: @JacisNonsense is this the correct conversion of the code you had originally?

File: ui/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+    }
+}
+
+fun setupTestSharedLib() {
```
Comment (JLLeitschuh) [inline ui/build.gradle.kts]: This whole thing needs to be tested. I couldn't get it to work on my machine.

File: build.gradle [hunk 3]
```diff
@@ -367,175 +365,74 @@ project(":ui") {
...
-        // This causes old mats (which take up little space on the heap but a lot of native memory) to get deallocated
-        // and free up native memory quicker, limiting the memory the app takes up.
-        jvmArgs = ["-XX:-OmitStackTraceInFastThrow", "-Xmx200m", "-XX:MaxNewSize=32m"]
-
-        bundleArguments = [
-                "linux.launcher.url": file('linuxLauncher/build/exe/linuxLauncher/linuxLauncher').toURI().toURL()
-        ]
-    }
-    mainClassName = jfx.mainClass
+//    task testSharedLib() {
```
Comment (JLLeitschuh) [inline build.gradle]: This task was never configured right. It actually does nothing and all of the `exec` statements would be run immediately (I think).

I left this here to help me figure out if I ported it wrong.

File: ui/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+    testCompile(files(coreOutput.classesDirs))
+    testCompile(files(coreOutput.resourcesDir))
+    testCompile(group = "org.testfx", name = "testfx-core", version = "4.0.5-alpha")
+    testCompile(group = "org.testfx", name = "testfx-junit", version = "4.0.5-alpha")
+    testRuntime(group = "org.testfx", name = "openjfx-monocle", version = "1.8.0_20")
+    testCompile(group = "org.opencv", name = "opencv-java", version = "3.1.0")
+}
+
+evaluationDependsOn(":core")
+evaluationDependsOn(":ui:preloader")
```
Comment (JLLeitschuh) [inline ui/build.gradle.kts]: I feel like this could be done better since I understand this better now.

File: ui/build.gradle.kts [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+                    exclude("**/ui/codegeneration")
+                    exclude("**/ui/codegeneration/**")
+                }
+            }
+        }
+    }
+}
+val coreProject = project(":core")
+dependencies {
+    compile(project(path = ":core", configuration = "shadow"))
```
Comment (JLLeitschuh) [inline ui/build.gradle.kts]: We could get rid of this extra level of complexity if we were to fully remove the deployer.

File: ui/build.gradle.kts (hunk 1 above)
Comment (JacisNonsense) [inline ui/build.gradle.kts]: I believe so. That should make it all available to IDEA

**Pull request closed without merging**